pynicotine/shares.py

```diff
@@ -744,6 +725,69 @@ def get_compressed_shares_message(self, share_type):
 
         return self.compressed_shares.get(share_type)
 
+    def get_shared_folders(self):
+
+        shared_folders = config.sections["transfers"]["shared"]
+        shared_folders_buddy = config.sections["transfers"]["buddyshared"]
```

Comment:
Need to verify if these have to be copied before passing them to the scanner process.

Reply:
A copy is required unless you intend to make immediate changes to the shares that cannot be Cancelled by destoying the Preferences dialog.

Reply:
We definitely want copies, but I'm curious if multiprocessing does some magic here already. I'll verify later.

Reply:
I don't understand why we still need to pass `shared_folders` around between functions within the shares module.

Reply:
> I suppose it is need while the dialog list box is in the process of being edited.

Yes.

The lists are indeed copied when passed to the scanner process already.

Reply:
The other thing I don't understand is why the database is not enough on it's own for storing the list of shared folders and virtual names. What is the benefit of having config settings for this?

Reply:
None of the current .db files are designed for storing a list of top-level shares, and the files are not human-readable/editable anyway. They are also more "temporary" than say the downloads/uploads.json files, in the sense that they're essentially caching folder/file data in a way that's convenient for Nicotine+'s needs, and can be rebuilt if they're lost.